caikit/runtime/http_server.py

```diff
@@ -237,16 +237,22 @@ async def _handler(
         ) -> Union[pydantic_response, Response]:
             log.debug("In unary handler for %s for model %s", rpc.name, model_id)
             loop = asyncio.get_running_loop()
-            request_kwargs = {
-                field: getattr(request, field) for field in request.__fields__
-            }
+            request_kwargs = dict(request)
+            required_params = rpc.task.get_required_parameters(False)
+            input_name = None
+            # handle required param input name
+            if len(required_params) == 1:
+                input_name = list(required_params.keys())[0]
             # flatten inputs and params into a dict
             # would have been useful to call dataobject.to_dict()
             # but unfortunately we now have converted pydantic objects
             combined_dict = {}
             for field in request_kwargs:
                 if request_kwargs[field]:
-                    combined_dict.update(**dict(request_kwargs[field]))
+                    if field == REQUIRED_INPUTS_KEY and input_name:
+                        combined_dict.update({input_name: request_kwargs[field]})
+                    else:
+                        combined_dict.update(**dict(request_kwargs[field]))
```

Comment:
to test this thoroughly, we should ideally have a module that has multiple required params?

Reply:
probably!
For now I'm fine with leaving things only tested with 1 required input parameter and punting to a discussion about what that api should look like. I bet it'll end up looking suspiciously like what you have implemented here, and in the meantime our stance can be "only tasks with 1 required input have HF-aligned rest support"